storage/engine: cleanup MapProvidingEngine interface

petermattis opened the pull request:
Remove the MapProvidingEngine interface which was not providing any
value as all of the users of MapProvidingEngine were only using the
diskmap.Factory interface.

Change the implementation of NewTempEngine to return a struct that
implements diskmap.Factory rather than having storage/engine.RocksDB
implement diskmap.Factory.

These changes pave the way for a non-RocksDB implementation of
diskmap.Factory via a temp engine.

Release note: None

petermattis committed:
storage/engine: cleanup MapProvidingEngine interface

Remove the MapProvidingEngine interface which was not providing any
value as all of the users of MapProvidingEngine were only using the
diskmap.Factory interface.

Change the implementation of NewTempEngine to return a struct that
implements diskmap.Factory rather than having storage/engine.RocksDB
implement diskmap.Factory.

These changes pave the way for a non-RocksDB implementation of
diskmap.Factory via a temp engine.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/38001)
<!-- Reviewable:end -->

petermattis committed:
storage/engine: unexport the RocksDBMap* types

Unexport the RocksDBMap* types which were not being used outside of
storage/engine.

Release note: None

petermattis commented:
Ping.

ajkr reviewed (approved):
LGTM, sorry for the delay.

> Reviewed 6 of 6 files at r1, 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38001#-:-LiKA0mo3wkXSef9u4jb:b-8wac67)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

ajkr commented:
OK I found there is a core-prs reviewer group in which apparently I was the only member: https://github.com/orgs/cockroachdb/teams/core-prs/members. Maybe leaving it will help me see relevant PR requests.

petermattis commented:
bors r=ajkr
